public/app/features/dashboard/state/initDashboard.ts

```diff
@@ -210,6 +224,22 @@ export function initDashboard(args: InitDashboardArgs): ThunkResult<void> {
     }
 
     // yay we are done
+    const dashboardId = dashDTO.dashboard.uid;
+    if (dashboardId) {
+      dashboard.panels.forEach((panel) => {
+        if (panel.panels) {
+          panel.panels.forEach((subPanel) => {
+            subPanel.targets.forEach((target) => {
+              logQueryOnDashboardLoaded(target, subPanel, !!panel.collapsed, dashboardId);
+            });
+          });
+        } else {
+          panel.targets.forEach((target) => {
+            logQueryOnDashboardLoaded(target, panel, !!panel.collapsed, dashboardId);
+          });
+        }
```

Comment:
Not a fan of the implementation here... Trying to think of a neater way to cater for nested panels.

Open to suggestions 😃

Reply:
Nit but can use recursivity here:
```ts
if (dashboardId) {
  const logQueryOnDashboardPanels = (panels) => panels.forEach(panel => panel.panels ? logQueryOnDashboardPanels(panel.panels) : panel.targets.forEach((target) => logQueryOnDashboardLoaded(target, panel, !!panel.collapsed, dashboardId)));
  logQueryOnDashboardPanels(dashboard.panels);
 }

Reply:
That's neater indeed , and would future proof the code in case we start adding sub-rows or something 🙂 

But then I would have to tweak it for the `collapsed` as it's using the parent's prop. The idea behind that prop was to give us an idea of queries not directly visible.

Although, I have been looking deeper into the nested panels and it seems there can only be nested panels for a dashboard row which is collapsed. For an extended row, the panels are brought one level up in the dashboard json.